### PR TITLE
api_delegate client|listen bugfix

### PR DESCRIPTION
**Pull request by o100ja**
I've noticed the bug while implementing a amqp transport plugin.
It seems that __parseConfig__ is not able to handle the argument parsing when invoking __api_client__ through __api_delegate__, which resulted in always returning the default options (_web_ transport).

The fix is located in commit 2b8dc0b, where it seems to me to be a common mistake (call vs apply). 

Once changed everything started working again

I also did some general code cleanup located in all the other commits 

I haven't seen any specific documentation regarding the changes to package.json (wouldn't hurt to include something more elaborate in the contributing section on senecajs.org) so I took the liberty of adding myself to the contributors and changed the NPM repository since I'll be using the forked version on the project I'm working on.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,7 +69,7 @@
   ],
   "repository": {
     "type": "git",
-    "url": "https://github.com/rjrodger/seneca.git"
+    "url": "git+ssh://git@github.com:Beats/seneca.git"
```
Comment (geek) [inline package.json]: will need to change this.  Please update to https://github.com/senecajs/seneca.git

**Comment by geek** [comment]
Looks good!  Any ideas on how to add tests to test the bug, to prevent regressions?

**Comment by o100ja** [comment]
Regarding the tests I'll have to think about it since Seneca needs much more elaborate testing than what is currently there especially, for transport.
I'll get back to you in a couple of days when I finish a segment on the project that I'm currently working on.

Also I have a question regarding the transport layer documentation, which is a bit vague on certain points that are of great importance from an architectural perspective. For instance:

### Horizontal Scaling
Let's say that we're using AMQP transport  for inter-service communication and we have a seneca process that exposes a set of functionalities **svcA** (eg, Identity Management Service) and we have a another two forked daemons **svcB1**, **svcB2** (eg. Notification Service). So the result on the MQ server is that we have the following _action_ queues.
* seneca_svcA_pinned_message_act
* seneca_svcB_pinned_message_act

If we do a _**svcA**.act(**svcB_pinned_message**)_ how is it handled by the forked processes?
1. The message is fanned-out to both svcB processe
2. The message is load-balanced (simple round-robin)
3. The situation will cause an error when the second _**svcB**.listen(**svcB_pinned_message**)_ is invoked.

I'll be working on it in the following week or two, depending on your answer :) 

For the project I'm working on (which is a FinTech platform) I have use-cases for both resolution 1 and resolution 2.

File: .gitignore [hunk 2]
```diff
@@ -9,10 +9,9 @@ node_modules*
 README.html
 *.off
 *-off
-.DS_Store
 npm-debug.log
```
Comment (geek) [inline .gitignore]: please add .DS_Store back, useful for OSX

**Comment by geek** [comment]
Please pull the latest from master... some changes made it in that make this non-mergable

**Comment by geek** [comment]
Landed in b043e38badbfc1a5a8b7d1d8a0d7eaf04f938103

**Pull request closed without merging**